src/app/core/service/rest.service.ts

```diff
@@ -1,19 +1,33 @@
 import { isPlatformBrowser, isPlatformServer } from '@angular/common';
-import { HttpClient, HttpHeaders } from '@angular/common/http';
-import { Inject, Injectable, PLATFORM_ID } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+import { Inject, Injectable, makeStateKey, PLATFORM_ID, TransferState } from '@angular/core';
 import { REQUEST } from '@nguniversal/express-engine/tokens';
 import { Observable, of } from 'rxjs';
 import { map, tap } from 'rxjs/operators';
 
+import { APP_CONFIG, AppConfig } from '../../app.config';
+
+export const REST_CACHE_TRANSLATE_STATE = makeStateKey<any>('REST_CACHE_TRANSLATE_STATE');
+
 @Injectable({
   providedIn: 'root',
 })
 export class RestService {
 
-  private cache: Map<string, any>;
+  private cache: any;
```

Comment:
I am not sure this is a good idea or not. This in-memory cache will grow per server node process and each initial page request will transfer the entire cache to rehydrate in the browser. Best case, each rehydrated client performs the same UI interactions to make use of the cache. Worst case, each rehydrated client never utilizes the cache.

Theoretically, each GET request cached will be transferred and rehydrated making for fastest performance after rehydration but slow initial state transfer.

Reply:
Selectively caching for state transfer would be ideal. Caching all async requests that are made for most if not all initial page requests.

Reply:
Currently, every deep link to a different view request will cache on server and be rehydrated by each client browser.

Reply:
To make for optimal demoing. Preload all pages intended to demo with a page refresh. Then next client rehydrate will have fast navigation through those pages.